src/components/snapping-scroll.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useRef, ReactNode, useEffect } from 'react';
+import { StyleSheet, View, Animated, NativeSyntheticEvent, NativeScrollEvent } from 'react-native';
+
+interface SnappingScrollProps {
+    scrollItems: ReactNode[],
+    scrollItemsHeight: number,
+    activeScrollItem?: number | null,
+    onScrollItemSnap: (index: number | null) => void,
+}
+
+const snappingScrollStyle = StyleSheet.create({
+    scrollViewContainer: {
+        paddingBottom: '50%',
+        paddingHorizontal: 10
+    },
+    itemContainer: {
+        alignItems: 'center',
+        justifyContent: 'center',
+    }
+});
+
+export const SnappingScroll: React.FC<SnappingScrollProps> = (props) => {
+
+    let currentScroll: number = 0;
+    const snappingAnimationMaxShrink = 0.95;
+    const snappingAnimationMaxOpacity = 0.6;
+    const snappingAnimationFade = 80;
+
+    const [animatedScrollValue] = useState(new Animated.Value(0));
+    const [scrollByUser, setScrollByUser] = useState(true);
+    const scrollRef = useRef<typeof Animated.ScrollView>(null);
+
+    useEffect(() => {
+        const activeScrollItem = props.activeScrollItem;
+        if (activeScrollItem !== null && activeScrollItem !== undefined) {
+            setScrollByUser(false);
+            scrollRef.current?.getNode().scrollTo({
+                y: activeScrollItem * props.scrollItemsHeight
+            });
+        }
+    }, [props.activeScrollItem])
+
+    const getItemScrollPos = (elementIndex: number) => {
+        return elementIndex * props.scrollItemsHeight;
+    }
+
+    const getItemIndexFromScrollPos = (scrollPos: number) => {
+        for (let i = 0; i < props.scrollItems.length; i++) {
+            if (getItemScrollPos(i) === scrollPos) {
+                return i;
+            }
+        }
+        return null;
+    }
```

Comment:
```suggestion
    const getItemIndexFromScrollPos = (scrollPos: number) => {
        for (let i = 0; i < props.scrollItems.length; i++) {
            if (getItemScrollPos(i) === scrollPos) {
                return i;
            } else if (scrollPos > getItemScrollPos(props.scrollItems.length-1)) {
                if(props.activeScrollItem === props.scrollItems.length-1) {
                    //Because the useEffect won't trigger if we are on the same item
                    setScrollByUser(false);
                    scrollRef.current?.getNode().scrollTo({
                        y: (props.scrollItems.length-1) * props.scrollItemsHeight
                    });
                }
                return (props.scrollItems.length-1);
            }
        }
        return null;
    }
```
It's currently possible to scroll past the last task, maybe do something like this to automatically focus on the last task if you scroll past it?